src/twisted/conch/ssh/filetransfer.py

```diff
@@ -35,28 +36,54 @@ def sendPacket(self, kind, data):
 
     def dataReceived(self, data):
         self.buf += data
-        while len(self.buf) > 5:
-            length, kind = struct.unpack("!LB", self.buf[:5])
+
+        # Continue processing the input buffer as long as there is a chance it
+        # could contain a complete request.  The "General Packet Format"
+        # (format all requests follow) is a 4 byte length prefix, a 1 byte
+        # type field, and a 4 byte request id.  If we have fewer than 4 + 1 +
+        # 4 == 9 bytes we cannot possibly have a complete request.
+        while len(self.buf) >= 9:
+            header = self.buf[:9]
+            length, kind, reqId = struct.unpack("!LBL", header)
+            # From the draft document:
```

Comment:
which draft document?

Reply:
Oops, referenced in a different comment I ended up deleting, I'll add the link here.